pkg/tenant/auth.go

```diff
@@ -13,6 +14,115 @@ import (
 // Set the maximum age of login protection cookies.
 const doubleCookiesMaxAge = time.Minute * 10
 
+// Register is a publically accessible endpoint that allows new users to create an
+// account via Quarterdeck by providing an email address and password.
+//
+// Route: POST /v1/register
+func (s *Server) Register(c *gin.Context) {
+	var err error
+
+	// Parse the request body
+	params := &api.RegisterRequest{}
+	if err = c.BindJSON(params); err != nil {
+		log.Warn().Err(err).Msg("could not parse request body")
+		c.JSON(http.StatusBadRequest, api.ErrorResponse("could not parse register request"))
+		return
+	}
+
+	// Validate that required fields were provided
+	if params.Name == "" || params.Email == "" || params.Password == "" || params.PwCheck == "" {
+		c.JSON(http.StatusBadRequest, api.ErrorResponse("missing required fields for registration"))
+		return
+	}
+
+	// Simple validation of the provided password
+	// Note: Quarterdeck also checks this along with password strength, but this allows
+	// us to filter some bad requests before they reach Quarterdeck.
+	if params.Password != params.PwCheck {
+		c.JSON(http.StatusBadRequest, api.ErrorResponse("passwords do not match"))
+		return
+	}
+
+	// Make the register request to Quarterdeck
+	req := &qd.RegisterRequest{
+		Name:     params.Name,
+		Email:    params.Email,
+		Password: params.Password,
+		PwCheck:  params.PwCheck,
+	}
+
+	// TODO: Handle error status codes returned by Quarterdeck
```

Comment:
The quarterdeck client returns a `StatusError` struct that can be used to get the status code from:

https://github.com/rotationalio/ensign/blob/main/pkg/quarterdeck/api/v1/errors.go#L103

Let's create a story for this TODO and mention the use of this error struct for better response handling.